crates/ruff_linter/src/rules/pyupgrade/rules/useless_class_metaclass_type.rs

```diff
@@ -0,0 +1,79 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Fix};
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+use ruff_python_ast::{self as ast, Expr, StmtClassDef};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use crate::fix::edits::{Parentheses, remove_argument};
+
+/// ## What it does
+/// Checks for `metaclass=type` in class definitions.
+///
+/// ## Why is this bad?
+/// Since Python 3, the default metaclass is `type`, so specifying it explicitly is redundant.
+///
+/// Even though `__prepare__` is not required, the default metaclass (`type`) implements it,
+/// for the convenience of subclasses calling it via `super()`.
+/// ## Example
+///
+/// ```python
+/// class Foo(metaclass=type): ...
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// class Foo: ...
+/// ```
+///
+/// ## References
+/// - [PEP 3115 – Metaclasses in Python 3000](https://peps.python.org/pep-3115/)
+#[derive(ViolationMetadata)]
+pub(crate) struct UselessClassMetaclassType {
+    name: String,
+}
+
+impl AlwaysFixableViolation for UselessClassMetaclassType {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let UselessClassMetaclassType { name } = self;
+        format!("Class `{name}` uses `metaclass=type`, which is redundant")
+    }
+
+    fn fix_title(&self) -> String {
+        "Remove `metaclass=type`".to_string()
+    }
+}
+
+/// UP050
+pub(crate) fn useless_class_metaclass_type(checker: &Checker, class_def: &StmtClassDef) {
+    let Some(arguments) = class_def.arguments.as_deref() else {
+        return;
+    };
+
+    for keyword in &arguments.keywords {
+        match (keyword.arg.as_deref(), &keyword.value) {
+            (Some("metaclass"), Expr::Name(ast::ExprName { id, .. })) if id == "type" => {
+                let mut diagnostic = Diagnostic::new(
+                    UselessClassMetaclassType {
+                        name: class_def.name.to_string(),
+                    },
+                    keyword.range(),
+                );
+
+                diagnostic.try_set_fix(|| {
```

Comment:
We shouldn't use `AlwaysFixable` in combination with `try_set_fix` because the try means that we sometimes fail to provide a fix